client/src/containers/NotRegisteredPage.tsx

```diff
@@ -70,103 +76,97 @@ export default class NotRegisteredPage extends Component<Props, State> {
       </span>
     );
 
+    const usersInputAddressFragment = usersInputAddress ? (
+      <>
+        {usersInputAddress.housenumber === " " ? "" : usersInputAddress.housenumber + " "}
+        {usersInputAddress.streetname !== " " && usersInputAddress.streetname}
+      </>
+    ) : null;
+
+    const registrationMissingOrExpired =
+      buildingInfo.registrationenddate === "" ||
+      getTodaysDate() > new Date(buildingInfo.registrationenddate);
+
+    const formattedLastRegDate = Helpers.formatDate(
+      buildingInfo.lastregistrationdate,
+      longDateOptions,
+      locale
+    );
+
+    const formattedRegEndDate = Helpers.formatDate(
+      buildingInfo.registrationenddate,
+      longDateOptions,
+      locale
+    );
+
+    const lastRegisteredCard = buildingInfo.lastregistrationdate ? (
+      <div className="card-body-registration text-center">
+        <p>
+          <b>
+            <Trans>Last registered:</Trans>
+          </b>{" "}
+          {formattedLastRegDate}
+          <span className="text-danger">
+            {" "}
+            <Trans>(expired {formattedRegEndDate})</Trans>
+          </span>
+        </p>
+      </div>
+    ) : (
+      <></>
+    );
+
+    let registrationMessage;
+
+    if (!usersInputAddress) {
+      registrationMessage = <Trans>No registration found!</Trans>;
+    } else if (buildingInfo.lastregistrationdate) {
```

Comment:
Hmmmm... so I just checked and it looks like there are 2200 hpd registrations where `lastregistrationdate` is NULL. Not totally sure what that means in the context of HPD's systems but I did also notice that `registrationenddate` is never NULL so maybe using that as our check here could be better? Up to you, as that may affect what we can assume about the truthiness of `lastregistrationdate` elsewhere in this component.

Reply:
I totally missed that, thanks for flagging that. I also updated things so this won't cause a weird null -> "1969-01-01" date issue when listing last registered dates.